### ISSUE TITLE
Bug Report: Game freezes after exiting freeplay menu

### ISSUE THREAD

**Issue by JackXson-Real** [FunkinCrew/Funkin#3178]
### Issue Checklist

- [X] I have properly named the issue
- [X] I have checked the issues/discussions pages to see if the issue has been previously reported

### What platform are you using?

Itch.io (Downloadable Build) - Windows

### If you are playing on a browser, which one are you using?

None

### Version

0.5.0

### Context (Provide images, videos, etc.)

Game freezes for a second after exiting freeplay menu. My video has no audio, but the freeze also stops the music from playing. You can see the freeze by looking at the animation of the text.

https://github.com/user-attachments/assets/580d17c4-066f-41fc-a54c-3ebe2802231c

### Steps to reproduce (or crash logs, errors, etc.)

Go into freeplay menu then exit.

**Comment by Hundrec** [comment]
Still happens in 0.6.3, that darn bug...

**Comment by JackXson-Real** [comment]
> Still happens in 0.6.3, that darn bug...

I think it has something to do with the memory leaks. The longer you stay in the menu, the longer it takes to exit.

**Comment by Hundrec** [comment]
I thought so, I'll link it to the issue

**Comment by NotHyper-474** [comment]
As I expected removing the bitmaps in a proper way (i.e storing their graphics in a var instead of inefficiently searching for them) solved the issue. 

https://github.com/user-attachments/assets/0f4141ed-d837-43b5-828f-362bce34decb

**Comment by JackXson-Real** [comment]
> As I expected removing the bitmaps in a proper way (i.e storing their graphic in a var to be destroyed later instead of inefficiently searching for it) solved the issue.
> 
>  2025-05-24.23-35-27.mp4

That menu music is wild lol. This definitely fixes the issue with it taking longer to load, but I don’t like how it still freezes. I remember like in 0.4.0 there was like no freeze whatsoever

**Comment by NotHyper-474** [comment]
> That menu music is wild lol. This definitely fixes the issue with it taking longer to load, but I don’t like how it still freezes. I remember like in 0.4.0 there was like no freeze whatsoever

Idk it feels the same for me? Here's a comparison of both 0.4.1 and a release build of 0.6.4

https://github.com/user-attachments/assets/df759ad4-fbd5-4e0d-9d55-03457ed81428

**Comment by JackXson-Real** [comment]
I double checked myself in 0.3.3 and yeah it seems like it was always there. Thanks for finding the fix @NotHyper-474, can you make a PR for it?

**Comment by NotHyper-474** [comment]
> I double checked myself in 0.3.3 and yeah it seems like it was always there. Thanks for finding the fix @NotHyper-474, can you make a PR for it?

Yeah though I would need to PR to either flxanimate or FunkinCrew's fork, so it might take it very long to be reviewed.
